test/battle/move_effect/octolock.c

```diff
@@ -0,0 +1,122 @@
+        {
+            MESSAGE("Foe Beldum can no longer escape because of Octolock!");
+            ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);
+            MESSAGE("Foe Beldum's Clear Body prevents stat loss!");
+            NOT MESSAGE("Foe Beldum's Defense fell!");
+            NOT MESSAGE("Foe Beldum's Sp. Def fell!");
+        }
+        else if (species == SPECIES_TORKOAL)
+        {
+            MESSAGE("Foe Torkoal can no longer escape because of Octolock!");
+            ABILITY_POPUP(opponent, ABILITY_WHITE_SMOKE);
+            MESSAGE("Foe Torkoal's White Smoke prevents stat loss!");
+            NOT MESSAGE("Foe Torkoal's Defense fell!");
+            NOT MESSAGE("Foe Torkoal's Sp. Def fell!");
+        }
+        else if (species == SPECIES_SOLGALEO)
+        {
+            MESSAGE("Foe Solgaleo can no longer escape because of Octolock!");
+            ABILITY_POPUP(opponent, ABILITY_FULL_METAL_BODY);
+            MESSAGE("Foe Solgaleo's Full Metal Body prevents stat loss!");
+            NOT MESSAGE("Foe Solgaleo's Defense fell!");
+            NOT MESSAGE("Foe Solgaleo's Sp. Def fell!");
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Octolock Defense reduction is prevented by Big Pecks")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_PIDGEY) { Ability(ABILITY_BIG_PECKS); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_OCTOLOCK); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_OCTOLOCK, player);
+        MESSAGE("Foe Pidgey can no longer escape because of Octolock!");
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent);
+        NOT MESSAGE("Foe Pidgey's Defense fell!");
+        ABILITY_POPUP(opponent, ABILITY_BIG_PECKS);
+        MESSAGE("Foe Pidgey's Big Pecks prevents Defense loss!");
+        MESSAGE("Foe Pidgey's Sp. Def fell!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Octolock reduction is prevented by Clear Amulet")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_CLEAR_AMULET); }
+    } WHEN {
+        TURN { MOVE(player, MOVE_OCTOLOCK); }
+        TURN {}
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_OCTOLOCK, player);
+        MESSAGE("Foe Wobbuffet can no longer escape because of Octolock!");
+        MESSAGE("Foe Wobbuffet's Clear Amulet prevents its stats from being lowered!");
+        NOT ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent);
+        NOT MESSAGE("Foe Wobbuffet's Defense fell!");
+        NOT MESSAGE("Foe Wobbuffet's Sp. Def fell!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Octolock will not decrease Defense and Sp. Def further then minus six")
+{
+    u8 j;
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_OCTOLOCK); }
+        for (j = 0; j < 6; j++)
+            TURN {}
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_OCTOLOCK, player);
+        for (j = 0; j < 5; j++)
```

Comment:
You're presumably missing curly brackets for the for loop here (unless you for some reason only want the Defense drop to be looped, in which case the Sp. Def drop needs an indentation less

Reply:
yep brackets were needed